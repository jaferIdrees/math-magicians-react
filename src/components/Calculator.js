import React, { useState } from 'react';
import calculate from '../logic/calculate';

const keys = ['AC', '+/-', '%', 'รท',
  7, 8, 9, 'x',
  4, 5, 6, '-',
  1, 2, 3, '+',
  0, '.', '='];

function CalculatorKey(props) {
  const props1 = { ...props };
  return (<input type="button" onClick={props1.clickHandler} className={(Number(props1.value) > 0) ? 'number' : 'key'} value={props1.value} />);
}

function keysElements(clickHandler) {
  return (keys.map((key) => (
    <CalculatorKey value={key} key={key.toString()} clickHandler={clickHandler} />)));
}

function Calculator() {
  const initialData = {};
  const [calculatorData, doCalculate] = useState(initialData);
  const clickHandler = (e) => {
    doCalculate((calculatorData) => calculate(calculatorData, e.target.value));
  };
  const keys = keysElements(clickHandler);
  const { next, total } = calculatorData;
  return (
    <div className="calculatorContainer">
      <input type="text" className="result" value={next || total || 0} disabled />
      {keys}
    </div>
  );
}

export default Calculator;
