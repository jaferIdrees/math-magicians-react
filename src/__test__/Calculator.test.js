import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Calculator component testing using snapshots', () => {
  it('Test Calculator component renders', () => {
    const tree = renderer
      .create(<Calculator />);
    expect(tree).toMatchSnapshot();
  });
});
