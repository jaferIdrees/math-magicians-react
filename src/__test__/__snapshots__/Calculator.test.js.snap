// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component testing using snapshots Test Calculator component renders 1`] = `
Array [
  <h2
    className="calculatorTile"
  >
    Let's do some math!
  </h2>,
  <div
    className="calculatorContainer"
  >
    <input
      className="result"
      disabled={true}
      type="text"
      value={0}
    />
    <input
      className="key"
      onClick={[Function]}
      type="button"
      value="AC"
    />
    <input
      className="key"
      onClick={[Function]}
      type="button"
      value="+/-"
    />
    <input
      className="key"
      onClick={[Function]}
      type="button"
      value="%"
    />
    <input
      className="key"
      onClick={[Function]}
      type="button"
      value="รท"
    />
    <input
      className="number"
      onClick={[Function]}
      type="button"
      value={7}
    />
    <input
      className="number"
      onClick={[Function]}
      type="button"
      value={8}
    />
    <input
      className="number"
      onClick={[Function]}
      type="button"
      value={9}
    />
    <input
      className="key"
      onClick={[Function]}
      type="button"
      value="x"
    />
    <input
      className="number"
      onClick={[Function]}
      type="button"
      value={4}
    />
    <input
      className="number"
      onClick={[Function]}
      type="button"
      value={5}
    />
    <input
      className="number"
      onClick={[Function]}
      type="button"
      value={6}
    />
    <input
      className="key"
      onClick={[Function]}
      type="button"
      value="-"
    />
    <input
      className="number"
      onClick={[Function]}
      type="button"
      value={1}
    />
    <input
      className="number"
      onClick={[Function]}
      type="button"
      value={2}
    />
    <input
      className="number"
      onClick={[Function]}
      type="button"
      value={3}
    />
    <input
      className="key"
      onClick={[Function]}
      type="button"
      value="+"
    />
    <input
      className="number"
      onClick={[Function]}
      type="button"
      value={0}
    />
    <input
      className="key"
      onClick={[Function]}
      type="button"
      value="."
    />
    <input
      className="key"
      onClick={[Function]}
      type="button"
      value="="
    />
  </div>,
]
`;
